From 17be07e230659c2b8b12e29211c5d24d0619d5ed Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Tue, 28 May 2013 17:16:01 +0200
Subject: [PATCH 1/2] configure: add option to disable building gtkmm-utils

* it looks strange in project named gtkmm-utils, but in some environments it's useful
  to build only glibmm-utils
---
Upstream-Status: Pending

 Makefile.am  |    8 ++++++--
 configure.ac |   18 +++++++++++++++---
 2 files changed, 21 insertions(+), 5 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 38b0e00..d6b8bf8 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,10 +1,14 @@
 ## top directory
 
 # Build in these directories:
-SUBDIRS = glibmm-utils gtkmm-utils examples
+SUBDIRS = glibmm-utils
+
+if GTKMM_UTILS_BUILD_GTKMM_UTILS
+    SUBDIRS += gtkmm-utils examples
+endif
 
 if GTKMM_UTILS_BUILD_DOCUMENTATION
-SUBDIRS += docs
+    SUBDIRS += docs
 endif
 
 EXTRA_DIST = AUTHORS COPYING HACKING INSTALL NEWS README autogen.sh configure.ac
diff --git a/configure.ac b/configure.ac
index e221387..76fb0d0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -154,6 +154,15 @@ AC_ARG_WITH([pkgconfig-dir],
     [pkgconfigdir='$(libdir)/pkgconfig'])
 AC_SUBST([pkgconfigdir])
 
+# glibmm-utils only?
+AC_MSG_CHECKING([gtkmm])
+AC_ARG_ENABLE([gtkmm],
+    AS_HELP_STRING([--enable-gtkmm],
+        [allows to disable gtkmm dependency and build only glibmm-utils (default: "True")]),
+    [enable_gtkmm="$enableval"],
+    [enable_gtkmm="yes"])
+AC_MSG_RESULT([$enable_gtkmm])
+
 # Check for dependencies and set build libs and cflags
 PKG_CHECK_MODULES([LIBGTHREAD], [gthread-2.0 >= 2.8.5])
 
@@ -165,10 +174,11 @@ PKG_CHECK_MODULES([GLIBMM], [glibmm-2.4 >= 2.12.0 sigc++-2.0])
 AC_SUBST(GLIBMM_LIBS)
 AC_SUBST(GLIBMM_CFLAGS)
 
-PKG_CHECK_MODULES([GTKMM], [gtkmm-2.4 >= 2.10.0])
+AS_IF([test "x$enable_gtkmm" != "xno"],
+    [PKG_CHECK_MODULES([GTKMM], [gtkmm-2.4 >= 2.10.0])
 
-AC_SUBST(GTKMM_LIBS)
-AC_SUBST(GTKMM_CFLAGS)
+    AC_SUBST(GTKMM_LIBS)
+    AC_SUBST(GTKMM_CFLAGS)])
 
 GTKMM_UTILS_LIBS="$LIBGTHREAD_LIBS $GLIBMM_LIBS $GTKMM_LIBS"
 GTKMM_UTILS_CFLAGS="$LIBGTHREAD_CFLAGS $GLIBMM_CFLAGS $GTKMM_CFLAGS"
@@ -213,6 +223,7 @@ AC_CONFIG_FILES(
 )
 fi
 AM_CONDITIONAL([GTKMM_UTILS_BUILD_DOCUMENTATION], [test "$enable_documentation" = "yes"])
+AM_CONDITIONAL([GTKMM_UTILS_BUILD_GTKMM_UTILS], [test "$enable_gtkmm" = "yes"])
 
 GTKMM_UTILS_INSTALL_PREFIX=$prefix
 
@@ -245,6 +256,7 @@ cat <<EOT
     C Compiler                      : ${CC}
     C++ Compiler                    : ${CXX}
     Build documentation             : ${enable_documentation}
+    Build gtkmm-utils               : ${enable_gtkmm}
     GTKMM_UTILS_DEVEL env variable  : ${GTKMM_UTILS_DEVEL}
     CFLAGS                          : ${CFLAGS}
     CXXFLAGS                        : ${CXXFLAGS}
-- 
1.7.10.4

